src/components/TextField/TextField.tsx

```diff
@@ -129,394 +120,356 @@ export type TextFieldProps = NonMutuallyExclusiveProps &
+
+  const className = classNames(
+    styles.TextField,
+    Boolean(normalizedValue) && styles.hasValue,
+    disabled && styles.disabled,
+    readOnly && styles.readOnly,
+    error && styles.error,
+    multiline && styles.multiline,
+    focus && styles.focus,
+  );
+
+  const inputType = type === 'currency' ? 'text' : type;
+
+  const prefixMarkup = prefix ? (
+    <div className={styles.Prefix} id={`${id}Prefix`} ref={prefixRef}>
+      {prefix}
+    </div>
+  ) : null;
+
+  const suffixMarkup = suffix ? (
+    <div className={styles.Suffix} id={`${id}Suffix`} ref={suffixRef}>
+      {suffix}
+    </div>
+  ) : null;
+
+  const characterCount = normalizedValue.length;
+  const characterCountLabel = intl.translate(
+    maxLength
+      ? 'Polaris.TextField.characterCountWithMaxLength'
+      : 'Polaris.TextField.characterCount',
+    {count: characterCount, limit: maxLength},
+  );
+
+  const characterCountClassName = classNames(
+    styles.CharacterCount,
+    multiline && styles.AlignFieldBottom,
+  );
+
+  const characterCountText = !maxLength
+    ? characterCount
+    : `${characterCount}/${maxLength}`;
+
+  const characterCountMarkup = showCharacterCount ? (
+    <div
+      id={`${id}CharacterCounter`}
+      className={characterCountClassName}
+      aria-label={characterCountLabel}
+      aria-live={focus ? 'polite' : 'off'}
+      aria-atomic="true"
+    >
+      {characterCountText}
+    </div>
+  ) : null;
+
+  const clearButtonMarkup =
+    clearButton && normalizedValue !== '' ? (
+      <button
+        testID="clearButton"
+        className={styles.ClearButton}
+        onClick={handleClearButtonPress}
+        disabled={disabled}
+      >
+        <VisuallyHidden>
+          {intl.translate('Polaris.Common.clear')}
+        </VisuallyHidden>
+        <Icon source={CircleCancelMinor} color="inkLightest" />
+      </button>
     ) : null;
 
-    const suffixMarkup = suffix ? (
-      <div className={styles.Suffix} id={`${id}Suffix`} ref={this.suffix}>
-        {suffix}
-      </div>
+  const handleNumberChange = useCallback(
+    (steps: number) => {
+      if (onChange == null) {
+        return;
+      }
+      // Returns the length of decimal places in a number
+      const dpl = (num: number) => (num.toString().split('.')[1] || []).length;
+
+      const numericValue = value ? parseFloat(value) : 0;
+      if (isNaN(numericValue)) {
+        return;
+      }
+
+      // Making sure the new value has the same length of decimal places as the
+      // step / value has.
+      const decimalPlaces = Math.max(dpl(numericValue), dpl(step));
+
+      const newValue = Math.min(
+        Number(max),
+        Math.max(numericValue + steps * step, Number(min)),
+      );
+
+      onChange(String(newValue.toFixed(decimalPlaces)), id);
+    },
+    [id, max, min, onChange, step, value],
+  );
+
+  const handleButtonRelease = useCallback(() => {
+    clearTimeout(buttonPressTimer.current);
+  }, [buttonPressTimer]);
```

Comment:
We can kill this as a dependency

```suggestion
  }, []);
```